openapi2/openapi2.go

```diff
@@ -179,14 +291,58 @@ type Header struct {
 type SecurityRequirements []map[string][]string
 
 type SecurityScheme struct {
-	Ref              string            `json:"$ref,omitempty"`
-	Description      string            `json:"description,omitempty"`
-	Type             string            `json:"type,omitempty"`
-	In               string            `json:"in,omitempty"`
-	Name             string            `json:"name,omitempty"`
-	Flow             string            `json:"flow,omitempty"`
-	AuthorizationURL string            `json:"authorizationUrl,omitempty"`
-	TokenURL         string            `json:"tokenUrl,omitempty"`
-	Scopes           map[string]string `json:"scopes,omitempty"`
-	Tags             openapi3.Tags     `json:"tags,omitempty"`
+	Extensions       map[string]interface{} `json:"-"`
+	Ref              string                 `json:"$ref,omitempty"`
+	Description      string                 `json:"description,omitempty"`
+	Type             string                 `json:"type,omitempty"`
+	In               string                 `json:"in,omitempty"`
+	Name             string                 `json:"name,omitempty"`
+	Flow             string                 `json:"flow,omitempty"`
+	AuthorizationURL string                 `json:"authorizationUrl,omitempty"`
+	TokenURL         string                 `json:"tokenUrl,omitempty"`
+	Scopes           map[string]string      `json:"scopes,omitempty"`
+	Tags             openapi3.Tags          `json:"tags,omitempty"`
+}
+
+func (securityScheme *SecurityScheme) UnmarshalJSON(b []byte) (err error) {
+	type _SecurityScheme SecurityScheme
+	s := _SecurityScheme{}
+	if err := json.Unmarshal([]byte(b), &s); err != nil {
+		return err
+	}
+	if err := json.Unmarshal([]byte(b), &s.Extensions); err != nil {
+		return err
+	}
+	StripNonCustomExtensions(&s.Extensions)
+	*securityScheme = SecurityScheme(s)
+	return nil
+}
+
+func (s SecurityScheme) MarshalJSON() ([]byte, error) {
+	return marshallFromMap(s.Extensions, reflect.ValueOf(s), reflect.TypeOf(s))
+}
+
+func StripNonCustomExtensions(extensions *map[string]interface{}) {
```

Comment:
I don't think you need to use a pointer to a map as map are already references, no?